### ISSUE TITLE
Problem with dd when using /dev/zero as input

### ISSUE THREAD

**Issue by abdulbadii** [msys2/MINGW-packages#22843]
### Package name

Bash

### For which environment do you need the package?

MINGW64

### Provide a brief description why you need this package in the selected environment

Apologize if it seem improper and unrelated to this place but can't help to ask a salvation from a painful confusing;
as Msys2/Mingw64 mainstream provide us dd beside Bash,

I've tried to use it to get a blank file;
`dd if=/dev/zero of=test.img bs=4M count=473`
which is incapable of

so research end up with http://www.chrysocome.net/downloads/ddrelease64.exe
one fork/version would perform it, with ddrelease64 name renamed to mere dd

But on applying it here, ie. in Msys2 environment, and as data or workspace in ext4 Linux' FS so move it by tool Ext2FS file manager, and did

./dd.exe if=//dev//zero' of=iso.img bs=4M count=473
./dd.exe if='\/dev\/zero' of=iso.img bs=4M count=473
./dd.exe if=\dev\zero of=iso.img bs=4M count=473
./dd.exe if=\\dev\\zero of=iso.img bs=4M count=473

..so on
by such literal CLI syntax which have failed too

So I strongly guess.. it seems the author does mean it'd work surely only on Windows' native console, so go for it click cmd shortcut,

H:>dd.exe if=/dev/zero' of=iso.img bs=4M count=473 

and.. correctly works perfect

So how could it be quite badly fail in Bash here even after ~10 CLI differebt syntax tried
How the real correct solution ?

$ ./dd.exe if=/dev/zero of=iso.img bs=4M count=473
rawwrite dd for windows version 1.0beta1 WIN64.
Written by John Newbigin <jnewbigin@chrysocome.net>
This program is covered by terms of the GPL Version 2.

Error opening input file: 3 The system cannot find the path specified.

Version
```
$ uname
MINGW64_NT-10.0-22621
```

### Are you willing to submit a PR?

no

**Comment by Kreijstal** [comment]
so you're saying dd from msys2 has a bug?

**Comment by striezel** [comment]
The hint is in the error message:

> Error opening input file: 3 The system cannot find the path specified.

The program (`dd`) cannot find the file `/dev/zero`. This is to be expected, because `/dev/zero` does not exist on Windows systems, it's a Linux / Unix feature. Therefore, like any other file that does not exist on Windows, `/dev/zero` cannot be used as input for dd.

But you may ask why it works with that ddrelease64.exe you downloaded from another site. Good question! That is because the author of that variant of dd has implemented `/dev/zero` as virtual device. Quoting from <http://www.chrysocome.net/dd>:

> Virtual devices
> 
> Virtual devices are a new feature in version 0.4beta1.
> 
> Because windows does not have devices like the unix `/dev/zero` or `/dev/random` these have been implemented inside dd. You can use these as input files to supply an infinite amount of zeros or pseudo random data.

So the only reason that works there is because this program treats `/dev/zero`  (and `/dev/random`) special by emulating those files as if they would exist. Which they don't on Windows.

In other words: This is not a bug in dd, it works as intended. Therefore I am closing this issue.
